test/graph/filter_validator.cpp

```diff
@@ -40,7 +42,13 @@ void hash_buffer(const AuditBuffer<T> &buf, unsigned p, unsigned width, unsigned
 
 	for (unsigned i = 0; i < height; ++i) {
 		const unsigned char *ptr = static_cast<const unsigned char *>(image_buffer[p][i]);
+#if (__BYTE_ORDER == __LITTLE_ENDIAN)
 		SHA1Update(&sha_ctx, ptr, width * sizeof(T));
+#else
+		for (unsigned j = 0; j < width; j++)
```

Comment:
Make an endian-reversed copy of the scanline instead.

Reply:
@sekrit-twc Hi, I am interested in reviving this PR as it seems to be the only thing standing in the way on big-endian.  I see this code has been reorganized into https://github.com/sekrit-twc/graphengine.  In this comment, do you mean that a copy should be made here, at hashing time, or do you mean that it should be done earlier, during actual buffer manipulations?

Reply:
I don't know. There aren't any semantics for big-endian unit tests at the moment.

Reply:
Github supports running on s390x natively:  https://github.com/marketplace/actions/run-on-architecture#supported-platforms

It would probably be nice to have a runner for all these architectures, it would help catch issues like https://github.com/sekrit-twc/zimg/issues/175

I'll post a PR that does byte-swapping during the checksum generation.  I don't think there's an easier way.